### ISSUE TITLE
Give keras.engine.training.Model its own np.random.RandomState

### ISSUE THREAD

**Issue by MaxG87** [keras-team/keras#8906]
When trying to make my Keras models reproducible, I noticed that I have to call np.random.seed right before invoking model.fit(..., shuffle=True, ...).

Checking the source code I noticed that model.fit finally invokes np.random.shuffle(...), thus altering the global random state. This can be fixed very easily by introducing a new member of type np.random.RandomState, and then using this one instead of the global one.

I would appreciate very much if this could be fixed as it would remove one source of irreproducibility.

**Comment by gabrieldemarmiesse** [comment]
Hi!
I don't really understand the expected behaviour. Can you give a script that put in evidence the problem you're facing? 
Thank you.

**Comment by MaxG87** [comment]
I've written a minimal working example which hopefully will clarify the problem.

I call np.random.seed() several times and check the first returned random number. If I call model.fit directly after reseeding, that number changes. In my opinion this should not happen.

```
#/usr/bin/python3

from tensorflow import set_random_seed
set_random_seed(1337)
import numpy as np
np.random.seed(1337)

from keras.layers import Dense
from keras.models import Sequential


# Generate some data
ndata = 10000
myseed = 99991
np.random.seed(myseed)
data = np.random.randn(ndata)
labels = data < 0.5

# Generate stub model
model = Sequential()
model.add(Dense(units=1, activation='sigmoid', input_shape=(1,)))
model.compile(loss='mse', optimizer='adam')

# Consume and reseed global randomness
np.random.seed(myseed)
ref_val = np.random.rand()
np.random.seed(myseed)
assert ref_val == np.random.rand()
np.random.seed(myseed)
assert ref_val == np.random.rand()
np.random.seed(myseed)
model.fit(data, labels, epochs=1, shuffle=True, verbose=0)
post_fit_val = np.random.rand()
print(ref_val, post_fit_val)
assert ref_val == post_fit_val
```

**Comment by gabrieldemarmiesse** [comment]
Thank you very much! This will help a lot. I'm reading it and making a PR if necessary.

**Comment by gabrieldemarmiesse** [comment]
I must admit that I'm not sure if this can be qualified as a bug. From my knowledge (quite limited, I don't have much experience), libraries don't care about altering the global random state. 

Though I don't know and might be wrong. I'll do the pull request putting what you said in evidence. Please defend your point of view in the pull request thread if necessary.

**Comment by MaxG87** [comment]
Thank you very much.
